mantidproject/mantid
# Update KafkaEventListener for new message schema

ISIS, PSI and ESS have agreed on message schema which should be suitable for use at the three institutions. These changes update KafkaEventListener for compatibility with these schema.

**To test:**
- Use [this docker-compose script](https://gist.github.com/matthew-d-jones/151602c00fc88b6e56c914797b33fcd6) to launch a Kafka broker and publish example data to it. You'll need docker and docker-compose installed (on ubuntu `apt install docker docker-compose`), then put the script somewhere and run `docker-compose up`
- Launch MantidPlot and set your default facility to TEST_LIVE
- Enter these options in the Live Data interface:
![livedata_options](https://user-images.githubusercontent.com/13455433/33766085-d2c894a6-dc13-11e7-9686-76dd95da376e.png)
- Hit run, you should end up with a workspace like this (compare the number of events):
![result](https://user-images.githubusercontent.com/13455433/33766152-29958d66-dc14-11e7-9e59-8e4eba0d430e.png)
- You may want to clean up the docker containers
```
docker-compose rm -v
```
and docker images, use `docker images` to list IDs of images, then
```
docker rm <IMAGE_ID>
```

Fixes #19358
Fixes #19487
Fixes #21389

**Release Notes** 

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
